File: frontend/src/listing/listing.ts
import { User } from "./user";

export class Listing {
    id: number;
    title: string;
    description: string;
    price: number;
    user: User;
    image_url: string;
    updated_date: string;
    pub_date: string;
  
    constructor(
      id: number,
      title: string,
      description: string,
      price: number,
      user: any,
      image_url: string,
      updated_date: string,
      pub_date: string
    ) {
      this.id = id;
      this.title = title;
      this.description = description;
      this.price = price;
      this.user = user;
      this.image_url = image_url;
      this.updated_date = updated_date;
      this.pub_date = pub_date;
    }
  }
  

File: backend/src/main/java/com/example/oferte_directe/entity/Listing.java
package com.example.oferte_directe.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
@Entity
@Table(name="listing")
@Data
public class Listing {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private int price;
    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;
    private String image_url;
    private LocalDateTime pub_date;
    private LocalDateTime updated_date;

}
